docs/template_plugin/tests/functional/op_reference/irdft.cpp

```diff
@@ -0,0 +1,765 @@
+    0.77598541,   0.46882716,   0.40922151,   0.28451272,   0.27156988,   0.32720683,   0.48740341,
+    0.52519462,   0.47371313,   0.61046947,   0.46505542,   0.04019986,   0.27622309,   0.42926452,
+    0.49897225,   0.04617115,   0.50902017,   0.74826910,   0.28548445,   0.63409441,   0.13183664,
+    0.02507987,   0.51695660,   0.50593892,   0.17335312,   0.24157819,   0.45513622,   0.69800550,
+    0.40604969,   0.47128647,   0.59389774,   0.33534107,   0.50887902,   0.82998967,   0.22642939,
+    0.32967160,   0.50515564,   0.54070049,   0.28947697,   0.35626388,   0.58235507,   0.30633221,
+    0.50041779,   0.24975602,   0.38320678,   0.40595842,   0.50651077,   0.42963483,   0.25977121,
+    0.32014694,   0.37577291,   0.46638206,   0.05511259,   0.45463482,   0.62685054,   0.13046773,
+    0.49768469,   0.47645129,   0.56182954,   0.74548830,   0.73150766,   0.37579758,   0.14279248,
+    0.28705593,   0.45403320,   0.50334282,   0.24132925,   0.24104091,   0.31220213,   0.62432518,
+    0.20954334,   0.09285936,   0.56852238,   0.42261752,   0.52830257,   0.25272655,   0.72091123,
+    0.46923499,   0.24439716,   0.72211522,   0.33004626,   0.30411236,   0.56189500,   0.37390448,
+    0.40768394,   0.13754946,   0.41746636,   0.50960175,   0.34250750,   0.65386079,   0.46042782,
+    0.54099804,   0.41183749,   0.40593833,   0.21666628,   0.38087729,   0.64666439,   0.19817938,
+    0.29519793,   0.46272810,   0.49454878,   0.59059650,   0.54134465,   0.56793991,   0.29395146,
+    0.52647797,   0.61291826,   0.24633402,   0.24791051,   0.22666050,   0.43238182,   0.20337301,
+    0.31388571,   0.59658993,   0.29774026,   0.39935257,   0.77171166,   0.54813165,   0.74253426,
+    0.49906203,   0.53449270,   0.22820431,   0.19888670,   0.56200754,   0.55242130,   0.36939947,
+    0.01671917,   0.60996081};
+
+template<class T>
+static std::vector<T> convert(const std::vector<float>& v) {
+    if (v.empty()) {
+        return std::vector<T>();
+    }
+
+    size_t num_of_elems = v.size();
+    std::vector<T> converted(num_of_elems);
+    for (size_t i = 0; i < num_of_elems; ++i) {
+        converted[i] = static_cast<T>(v[i]);
+    }
+    return converted;
+}
+
+template <class T>
+static std::vector<T> convert(const std::vector<float16>& v) {
+    if (v.empty()) {
+        return std::vector<T>();
+    }
+
+    size_t num_of_elems = v.size();
+    std::vector<T> converted(num_of_elems);
+    for (size_t i = 0; i < num_of_elems; ++i) {
+        converted[i] = static_cast<T>(v[i]);
+    }
+    return converted;
+}
+
+template <class T>
+static std::vector<T> convert(const std::vector<bfloat16>& v) {
+    if (v.empty()) {
+        return std::vector<T>();
+    }
+
+    size_t num_of_elems = v.size();
+    std::vector<T> converted(num_of_elems);
+    for (size_t i = 0; i < num_of_elems; ++i) {
+        converted[i] = static_cast<T>(v[i]);
+    }
+    return converted;
+}
```

Comment:
Could these three function templates be merged into one? They look identical (apart from input vector type).

Reply:
it looks like the `convert` method is no more needed ;) removed